### PR TITLE
Cache dependencies in GitHub Actions CI workflow

### PR DESCRIPTION
**Pull request by SergeAstapov**
PR to speed up CI times on GitHub Actions by caching dependencies.

References:
* https://github.com/actions/setup-node#caching-packages-dependencies
* https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon/files/.github/workflows/ci.yml

Main changes:
* introduces "Floating Dependencies" job
* uses `actions/setup-node@v2` with `cache: yarn` option

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -11,34 +11,49 @@ on:
 
 jobs:
   test:
-    name: Tests
+    name: "Tests"
```
Comment (scalvert) [inline .github/workflows/ci-build.yml:14]: Does this need to be quoted?

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -11,34 +11,49 @@ on:
...
+        run: yarn test
+
+  try-scenarios:
+    name: ${{ matrix.try-scenario }}
+    runs-on: ubuntu-latest
+    needs: 'test'
 
     strategy:
-      fail-fast: true
+      fail-fast: false
```
Comment (scalvert) [inline .github/workflows/ci-build.yml:54]: Why do we not want to fail fast?

**Review by scalvert — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (SergeAstapov) [reply .github/workflows/ci-build.yml:54]: This aligns with latest addon blueprint and more details in https://github.com/ember-cli/ember-cli/pull/9661

if you disagree - I can revert

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (SergeAstapov) [reply .github/workflows/ci-build.yml:14]: Not sure tbh, I think works either way, but better revert to align with addon blueprint

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (scalvert) [reply .github/workflows/ci-build.yml:54]: Cool. I'm fine with aligning with the default!

**Pull request merged**